apps/sms/views/conversation/js/conversation.js

```diff
@@ -2504,26 +2505,21 @@ var ConversationView = {
     // force a number
     var id = +messageDOM.dataset.messageId;
     var iccId = messageDOM.dataset.iccId;
-
-    var request = MessageManager.retrieveMMS(id);
-
     var button = messageDOM.querySelector('button');
 
     this.setMessageStatus(id, 'pending');
     button.setAttribute('data-l10n-id', 'downloading-attachment');
 
-    request.onsuccess = (function retrieveMMSSuccess() {
+    MessagingClient.retrieveMMS(id).then((message) => {
```

Comment:
__nit:__ we don't have ```message``` as return result, so better to remove to avoid confusion.